src/reference.rs

```diff
@@ -0,0 +1,76 @@
+use std::fmt::{Debug, Display, Formatter};
+use std::sync::Arc;
+
+use uuid::Uuid;
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
+pub struct Reference(Arc<String>);
+impl Reference {
+    pub fn new() -> Self {
+        Self::prefixed("")
+    }
+
+    fn prefixed(prefix: &str) -> Self {
+        Self(Arc::new(format!(
+            "{}{}",
+            prefix,
+            Uuid::new_v4()
+                .hyphenated()
+                .encode_upper(&mut Uuid::encode_buffer())
+        )))
+    }
+
+    pub fn heartbeat() -> Self {
+        Self::prefixed("heartbeat:")
+    }
+}
+impl Debug for Reference {
```

Comment:
This can be derived

Reply:
The derived version of Debug with alternate true, that is the {:#?} format specificier produces a pretty-printed format that still contains the name of the struct instead of just being the internal string.  I want the internal string only because it makes the hashes in the `State` cleaner to read when debugging:

```rust
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn reference_debug_without_alternate() {
        let reference: Reference = "reference".into();

        assert_eq!(format!("{:?}", reference), "Reference(\"reference\")")
    }

    #[test]
    fn reference_debug_with_alternate() {
        let reference: Reference = "reference".into();

        assert_eq!(format!("{:#?}", reference), "\"reference\"")
    }
}
```

```
thread 'reference::tests::reference_debug_with_alternate' panicked at 'assertion failed: `(left == right)`
  left: `"Reference(\n    \"reference\",\n)"`,
 right: `"\"reference\""`', src/reference.rs:95:9
```

So, the `Debug` as written is not equivalent to the derived one.